tests/end2end-tests/data_process/data_gen.py

```diff
@@ -131,6 +136,17 @@ def split_data(data, num):
             },
         ],
     },
+    {
```

Comment:
What will happen if the block of node1 with `"node_id_col": "id"` comes after the one without `"node_id_col": "id"`? For example
```
node_conf = [
   {
       "node_type": "node1",
        "format": {"name": "hdf5"},
        "files": os.path.join(in_dir, "node_data1_2.hdf5"),
        ...
    },
    {
        "node_id_col": "id",
        "node_type": "node1",
        "format": {"name": "parquet"},
        "files": os.path.join(in_dir, "node_data1_*.parquet"),
        "features": [
        ...
    }
```


Will the graph construction crash?

Reply:
i think it should be fine. let me modify the integration test to test this case.